uwblueprint/jump-math
# (5) feat: fetch classroom drilldown data

## Notion ticket link

[Fetch student data for class](https://www.notion.so/uwblueprintexecs/Fetch-student-data-for-class-b6b2e9bf209d4aa7953158d4c8d217fb)
[Fetch assessment data for class](https://www.notion.so/uwblueprintexecs/Fetch-assessment-data-for-class-8269112493c242239c0dc981ead86f6f)

_Reviewing by commit might make the PR easier to review!_


## Implementation description
- [feat: add class resolver to get classes by ID](https://github.com/uwblueprint/jump-math/pull/427/commits/737c91c625d52283f686656f55babad6b570c4c1)
- [feat: class client queries](https://github.com/uwblueprint/jump-math/pull/427/commits/adee0ae06048b18eda93fdf19a91e276f922a2c9)
- [feat: load assessment data](https://github.com/uwblueprint/jump-math/pull/427/commits/73c390f659252c7b0841628d139a70ed5f9b7ce3)
- [feat: load student data](https://github.com/uwblueprint/jump-math/pull/427/commits/7857def6dd8bd5d9ade1a7a82150fb8a1f74a8f3)
- [fix: sorting functions should copy their input, not mutate it](https://github.com/uwblueprint/jump-math/pull/427/commits/eb115c0ef452aaa28506b70419970887d29cca3c) 
- [fix: students should be re-fetched after adding a student](https://github.com/uwblueprint/jump-math/pull/427/commits/6882a48034c0f7d9dfd56ecf1f759e3dc8613cd1)
- [feat: fetch class title and start date](https://github.com/uwblueprint/jump-math/pull/427/commits/d9217e6e5c6e0b32c0ac80fa771d41a936234fa1)


<img width="1392" alt="image" src="https://github.com/uwblueprint/jump-math/assets/9922514/9c1885fe-4fc4-4aec-8382-8a179c484c97">




## Steps to test
1. View the class list page.
2. Click on a class card.
3. Verify all tables on the resulting page are fully populated with live data, and the data reflects the objects that actually exist.



## What should reviewers focus on?
* Are there any edge cases I missed?


## Checklist
